cai_causal_graph/time_series_causal_graph.py

```diff
@@ -289,36 +289,56 @@ def get_minimal_graph(self) -> TimeSeriesCausalGraph:
     def _get_time_topological_order(self, ordered_nodes: List[str]) -> List[str]:
         """Return the provided list if it is ordered in time; otherwise, an empty list is returned."""
-
         # Iterate through the list to ensure it respects the time ordering.
         for j, node in enumerate(ordered_nodes):
             if j == 0:
                 continue
             assert isinstance(node, str)  # for linting
             # check if the time delta is correct
-            if self.get_node(ordered_nodes[j - 1]).time_lag > self.get_node(ordered_nodes[j]).time_lag:   # type: ignore
+            if self.get_node(ordered_nodes[j - 1]).time_lag > self.get_node(ordered_nodes[j]).time_lag:  # type: ignore
                 return []
 
         return ordered_nodes
 
-    def get_topological_order(self, return_all: bool = False) -> Union[List[str], List[List[str]]]:
+    def get_topological_order(
+        self, return_all: bool = False, respect_time_ordering: bool = True
+    ) -> Union[List[str], List[List[str]]]:
         """
         Return the topological order of the graph that is ordered in time.
 
+        :param return_all: If `True`, return all the possible topological orders. Default is `False`.
+        :param respect_time_ordering: If `True`, return the topological order that is ordered in time. Default is
+            `True`.
+
         For more details, see `cai_causal_graph.causal_graph.CausalGraph.get_topological_order`.
         """
-        ordered_nodes = super().get_topological_order(return_all=True)
-        top_order_list: List[List[str]] = []
-        for current_top_order in ordered_nodes:
-            assert isinstance(current_top_order, list)  # for linting
-            new_order = self._get_time_topological_order(current_top_order)
-            if new_order not in top_order_list and len(new_order) > 0:
-                top_order_list.append(new_order)
-
-        if not return_all:
-            top_order_list = top_order_list[0]  # type: ignore
-
-        return top_order_list
+        if respect_time_ordering:
+            if return_all:
+                ordered_nodes_list = super().get_topological_order(return_all=return_all)
+                # ordered_nodes_list must be a list of lists of strings
+                assert isinstance(ordered_nodes_list, list)
+                assert all(isinstance(x, list) for x in ordered_nodes_list)
+
+                ordered_nodes: List[List[str]] = []
+
+                for i in range(len(ordered_nodes_list)):
+                    assert isinstance(ordered_nodes_list[i], list)
+                    tmp = self._get_time_topological_order(ordered_nodes_list[i])  # type: ignore
+                    if tmp not in ordered_nodes and len(tmp) > 0:
+                        ordered_nodes.append(tmp)
+
+                return ordered_nodes
+            else:
+                # check it is a dag
+                assert self.is_dag(), 'The graph is not a DAG. The topological order is not valid.'
```

Comment:
I think this always needs to be checked so would do it after `if respect_time_ordering` but before the next `if` block. No need to do in the `else` that calls `super` as it is checked in `super`.

Reply:
why? when `return_all=True` it will be checked by `super().get_topological_order(return_all=return_all)` if I am not mistaken

Reply:
Good point. I missed that `super` was called there as well. My bad.

Reply:
Would just add a comment that this check is not needed in other conditionals as it is done by `super`.